### PR TITLE
TLS DER secret transformer

### PR DESCRIPTION
**Pull request by seuf**

### CODE REVIEW

**Comment by maelvls** [comment]
Very nice! Do you think we could have both support for PEM and DER formats? I was thinking about using the extension (pem, der, p12, or pks) in order to choose which encoding should be used.

Examples:

```yaml
kind: Secret
metadata:
  annotations:
    secret-transform-key: tls.pem
    secret-transform-encoding: pem
```

and

```yaml
kind: Secret
metadata:
  annotations:
    secret-transform-key: tls.der
    secret-transform-encoding: der
```

What do you think?

File: main.go [hunk 1]
```diff
@@ -36,16 +36,16 @@ import (
 )
 
 const (
-	secretAnnotKey = "cert-manager.io/secret-transform"
-	tlsPEMDataKey  = "tls.pem"
+	secretAnnotKey = "cert-manager-secret-transform"
+	tlsDERDataKey  = "tls.der"
```
Comment (maelvls) [inline main.go:40]: Was this change required due to some limitation? Just wondering :grin:

Comment (seuf) [reply main.go:40]: > Very nice! Do you think we could have both support for PEM and DER formats? I was thinking about using the extension (pem, der, p12, or pks) in order to choose which encoding should be used.

I currently [working to add cert-manager additional key format support](https://github.com/jetstack/cert-manager/pull/4598).
The idea is to add both `PEM` and `DER` format support.
